### PR TITLE
fix(curriculum): allow multiple solutions oop game 73

### LINKED ISSUE #53115: Fix tests for OOP game step 73 to accept multiple valid solutions
### Describe the Issue

We had a camper on the [forum](https://forum.freecodecamp.org/t/learn-intermediate-oop-by-building-a-platformer-game-step-73/663187), unable to pass the test with the following code:

```js
platforms.forEach((platform) => platform.draw());
```
The test is expecting this solution

```js
 platforms.forEach((platform) => {
    platform.draw();
  });
```

We shouldn't have one fixed solution.

All valid solutions should be accepted.
The tests for this step will need to be updated 

### Affected Page

https://www.freecodecamp.org/learn/javascript-algorithms-and-data-structures-v8/learn-intermediate-oop-by-building-a-platformer-game/step-73

### Your code

see explanation above

### Expected behavior

see explanation above

### Screenshots

_No response_

### System

- Device: [e.g. iPhone 6, Laptop]
- OS: [e.g. iOS 14, Windows 10, Ubuntu 20.04]
- Browser: [e.g. Chrome, Safari]
- Version: [e.g. 22]

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by a2937**
Checklist:

- [X] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [X] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request).
- [X] My pull request targets the `main` branch of freeCodeCamp.
- [X] I have tested these changes either locally on my machine, or GitPod.

Closes #53115

All I had to do was make the braces optional.

### CODE REVIEW

**Review by jdwilkin4 — CHANGES_REQUESTED** [review]
We should also make it possible to pass with this code

```js
platforms.forEach(platform => platform.draw());
```
Screenshot 2024-01-10 at 5 56 33 PM

**Review by gikf — COMMENTED** [review]
Once I figure out reliable regex to match only when both `{}` are used or none, I'm going to be fully advocating doing that. For now just the suggestion to allow parameter both with and without parenthesis.

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-intermediate-oop-by-building-a-platformer-game/64c764dd9071050d0a2c1473.md [hunk 1]
```diff
@@ -18,19 +18,19 @@ Inside the callback function, add a `platform` parameter and for the body of the
 You should have a `forEach` loop that iterates through the `platforms` array.
 
 ```js
-assert.match(code, /platforms\.forEach\(\s*(.*?)\s*=>\s*{\s*(.*?)\s*}\s*\);?/);
+assert.match(code, /platforms\.forEach\(\s*(.*?)\s*=>\s*{?\s*(.*?)\s*}?\s*\);?/);
```
[the remark this answers is not in the document]
Comment (a2937) [reply curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-intermediate-oop-by-building-a-platformer-game/64c764dd9071050d0a2c1473.md]: Thank you for your suggestions. I applied them.

**Comment by JoyShaheb** [comment]
looks good to me but update your merge conflicts and please check for the unit tests. Good work !

**Review by JoyShaheb — APPROVED** [review]

**Review by jdwilkin4 — APPROVED** [review]
The changes work for me.

Once the other issues mentioned are resolved, then this will be good to go in

**Pull request merged**